### PR TITLE
docs: Update roadmap for the 1.14 release

### PR DESCRIPTION
**Pull request by travisn**
Looking forward to the 1.14 release in April, update the roadmap of planned feature work. The features are not limited to this list, it is just for context on what is currently planned.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -8,21 +8,21 @@ We hope that the items listed below will inspire further engagement from the com
...
+* Add build support for Go 1.22 [#13738](https://github.com/rook/rook/pull/13738)
 
 ## Kubectl Plugin
 
-Features are planned in the 1.13 time frame for the [Kubectl Plugin](https://github.com/rook/kubectl-rook-ceph).
-  * Recover the CephCluster CR after accidental deletion [#68](https://github.com/rook/kubectl-rook-ceph/issues/68)
-  * Force cleanup the cluster if graceful uninstall is not desired [#131](https://github.com/rook/kubectl-rook-ceph/issues/131)
-  * Provide a restricted set of commands based on a build flag [#174](https://github.com/rook/kubectl-rook-ceph/issues/174)
-  * Collect details to help troubleshoot the csi driver [#69](https://github.com/rook/kubectl-rook-ceph/issues/69)
+Features are planned in the 1.14 time frame for the [Kubectl Plugin](https://github.com/rook/kubectl-rook-ceph).
```
Comment (subhamkrai) [inline ROADMAP.md:26]: good to add issue https://github.com/rook/kubectl-rook-ceph/issues/74  for tracking?

File: ROADMAP.md [hunk 2]
```diff
@@ -8,21 +8,21 @@ We hope that the items listed below will inspire further engagement from the com
...
 See the [GitHub project boards](https://github.com/rook/rook/projects) for the most up-to-date issues and their status.
 
-## Rook Ceph 1.13
+## Rook Ceph 1.14
 
-The following high level features are targeted for Rook v1.13 (November 2023). For more detailed project tracking see the [v1.13 board](https://github.com/rook/rook/projects/30).
+The following high level features are targeted for Rook v1.14 (April 2024). For more detailed project tracking see the [v1.14 board](https://github.com/rook/rook/projects/31).
 
-* OSD encryption on partitions [#10984](https://github.com/rook/rook/issues/10984)
+* Allow setting the application name on a CephBlockPool [#13744](https://github.com/rook/rook/pull/13744)
```
Comment (parth-gr) [inline ROADMAP.md:15]: 1) add replica-1 support for external clusters
And internal cluster documentation

2) Support for service accounts

**Comment by rkachach** [comment]
Enable rook orchestrator by default: https://github.com/rook/rook/issues/13760

File: ROADMAP.md (hunk 1 above)
Comment (travisn) [reply ROADMAP.md:26]: Until someone is committed to that complex feature, I'll suggest to not include it in the feature list for a specific release.

File: ROADMAP.md [hunk 3]
```diff
@@ -8,21 +8,22 @@ We hope that the items listed below will inspire further engagement from the com
...
-The following high level features are targeted for Rook v1.13 (November 2023). For more detailed project tracking see the [v1.13 board](https://github.com/rook/rook/projects/30).
+The following high level features are targeted for Rook v1.14 (April 2024). For more detailed project tracking see the [v1.14 board](https://github.com/rook/rook/projects/31).
 
-* OSD encryption on partitions [#10984](https://github.com/rook/rook/issues/10984)
-* Object Store
-  * Pool sharing for clusters where many object stores are required [#11411](https://github.com/rook/rook/issues/11411)
-* CephFS
-  * Automatic subvolume group pinning [#12607](https://github.com/rook/rook/issues/12607)
-* Ceph-CSI [v3.10](https://github.com/ceph/ceph-csi/issues?q=is%3Aopen+is%3Aissue+milestone%3Arelease-v3.10.0)
+* Allow setting the application name on a CephBlockPool [#13744](https://github.com/rook/rook/pull/13744)
```
Comment (BlaineEXE) [inline ROADMAP.md:15]: Something like this to note our intended removal of the "holder" pattern.
 
```
* Improve upgrade workflow for future Multus-enabled clusters (begin "holder" pod deprecation) [#13055](https://github.com/rook/rook/issues/13055)
```

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**